clang/include/clang/Sema/ParsedAttr.h

```diff
@@ -1105,16 +1120,16 @@ enum AttributeDeclKind {
 
 inline const StreamingDiagnostic &operator<<(const StreamingDiagnostic &DB,
                                              const ParsedAttr &At) {
-  DB.AddTaggedVal(reinterpret_cast<uint64_t>(At.getAttrName()),
+  const IdentifierInfo *AttrName =
+      At.printAsMacro() ? At.getMacroIdentifier() : At.getAttrName();
+  DB.AddTaggedVal(reinterpret_cast<uint64_t>(AttrName),
```

Comment:
Instead of printing the macro name here, do we want to add a new tag `ak_attr` that takes a `ParsedAttr *` so that the diagnostics engine can automatically print both the macro name and the attribute name (with an aka)? Then we can suppress the follow-on note that lists macro expansions (maybe? If not, then maybe this is a bad idea because we're duplicating the information we show the user.)

Reply:
I like the idea of printing the attribute name as well in the same message (or in a note), I think that is valuable.

Reply:
@erichkeane Currently, a note is emitted to show the line where it's expanded from that includes the original attribute spelling:  https://github.com/llvm/llvm-project/pull/107619/files#diff-3d962c5c4f372776e40887d273f9404efe3cd306f9915505d8566b791021914bR49

Not sure how I can suppress the follow-on note that lists macro expansions, but I can try the `ak_attr` approach as @AaronBallman suggested.

Reply:
@AaronBallman @erichkeane Thanks for the valuable feedback!

Our motivation was that with `-fbounds-safety`, we would have a toolchain header defines this:
`#define __counted_by(c) __attribute__((__counted_by(c)__))` 

then the user code uses it as `__counted_by` spelling, 

we'd like it to be just printed as `__counted_by` and we don't want it to be printed as `__counted_by__` nor having a redundant aka (`__counted_by__` ) next to `__counted_by`.

Also, that was why we were limiting this change specifically to "system-defined" macros.

Does it make sense? I guess there might be a better way to implement this behavior in Clang?

Reply:
How about always printing the macro name (if exists) and printing "(aka 'foo')" only when it's not system-defined macro?